azurerm/internal/tools/website-scaffold/main_test.go

```diff
@@ -0,0 +1,123 @@
+package main
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/sergi/go-diff/diffmatchpatch"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/tags"
+)
+
+const (
+	RESOURCE_NAME    = "azurerm_foobar"
+	BRAND_NAME       = "Foobar"
+	RESOURCE_ID      = "12345"
+	WEBSITE_CATEGORY = "Foobar Category"
+)
+
+func setupDocGen(isDataSource bool, resource *schema.Resource) documentationGenerator {
+	return documentationGenerator{
+		resourceName: RESOURCE_NAME,
+		brandName:    BRAND_NAME,
+		resourceId: func(s string) *string {
+			return &s
+		}(RESOURCE_ID),
+		isDataSource:      isDataSource,
+		websiteCategories: []string{WEBSITE_CATEGORY},
+		resource:          resource,
+	}
+}
+
+func TestResourceArgumentBlock(t *testing.T) {
+	expectedOut := fmt.Sprintf(`## Arguments Reference
+
+The following arguments are supported:
+
+* %[1]sblock%[1]s - (Required) A %[1]sblock%[1]s block as defined below.
+
+* %[1]sfoo_enabled%[1]s - (Required) Should the TODO be enabled?
+
+* %[1]sfoo_id%[1]s - (Required) The ID of the TODO.
+
+* %[1]slist%[1]s - (Required) Specifies an array of TODO.
+
+* %[1]slocation%[1]s - (Required) The Azure Region where the Foobar should exist. Changing this forces a new Foobar to be created.
+
+* %[1]smap%[1]s - (Required) Specifies a map of TODO.
+
+* %[1]sname%[1]s - (Required) The Name which should be used for this Foobar. Changing this forces a new Foobar to be created.
+
+* %[1]sresource_group_name%[1]s - (Required) The name of the Resource Group where the Foobar should exist. Changing this forces a new Foobar to be created.
+
+* %[1]sset%[1]s - (Required) Specifies an array of TODO.
+
+---
+
+* %[1]stags%[1]s - (Optional) A mapping of tags which should be assigned to the Foobar.
+
+---
+
+A %[1]sblock%[1]s block supports the following:`, "`")
```

Comment:
My intention is to avoid use any char possible to appear here. While we have full control about what appears here, so it make sense to use `'` for clearer template.